### PR TITLE
Wait for async savings and removals

### PR DESCRIPTION
**Pull request by vepasto**

### CODE REVIEW

**Comment by vepasto** [comment]
This is combined pull request. 

Removals in article could be done in parallel, then I suggest to make mongoose async by using bluebird promisify. Would that be okay?

**Comment by vepasto** [comment]
Actually I had some randomly appearing errors caused by this issue. When I made this fix all to my tests those errors disappeared.

File: modules/users/tests/server/user.server.model.tests.js [hunk 1]
```diff
@@ -69,7 +70,8 @@ describe('User Model Unit Tests:', function() {
 	});
 
 	after(function(done) {
-		User.remove().exec();
-		done();
+		User.remove().exec(function(){
```
Comment (ilanbiala) [inline modules/users/tests/server/user.server.model.tests.js]: Can't you pass `done` to .exec?

**Comment by vepasto** [comment]
Now done is passed to exec as a parameter

**Comment by ilanbiala** [comment]
Same fixes here please.

**Comment by vepasto** [comment]
It seems to be already okay

```
Running "jshint:all" (jshint) task
>> 79 files lint free.
```

File: modules/articles/tests/server/article.server.model.tests.js [hunk 2]
```diff
@@ -57,8 +57,8 @@ describe('Article Model Unit Tests:', function() {
 	});
 
 	afterEach(function(done) {
-		Article.remove().exec();
-		User.remove().exec();
-		done();
+		Article.remove().exec(function(){
```
Comment (ilanbiala) [inline modules/articles/tests/server/article.server.model.tests.js]: space between `()` and `{`.

File: modules/articles/tests/server/article.server.routes.tests.js [hunk 3]
```diff
@@ -270,8 +270,8 @@ describe('Article CRUD tests', function() {
 	});
 
 	afterEach(function(done) {
-		User.remove().exec();
-		Article.remove().exec();
-		done();
+		User.remove().exec(function(){
```
Comment (ilanbiala) [inline modules/articles/tests/server/article.server.routes.tests.js]: space between `()` and `{`.

File: modules/users/tests/server/user.server.model.tests.js [hunk 4]
```diff
@@ -52,10 +52,11 @@ describe('User Model Unit Tests:', function() {
 		});
 
 		it('should fail to save an existing user again', function(done) {
-			user.save();
-			return user2.save(function(err) {
-				should.exist(err);
-				done();
+			user.save(function(){
```
Comment (ilanbiala) [inline modules/users/tests/server/user.server.model.tests.js]: space between `()` and `{`.

**Comment by vepasto** [comment]
Fixed formatting

**Comment by ilanbiala** [comment]
Great! @vepasto can you squash the commits down to 1 commit so that we don't pollute the commit log?

**Comment by vepasto** [comment]
Done

**Comment by vepasto** [comment]
Done

**Pull request merged**